src/mavis/convert/vcf.py

```diff
@@ -122,6 +123,99 @@ def parse_bnd_alt(alt: str) -> Tuple[str, int, str, str, str, str]:
         raise NotImplementedError('alt specification in unexpected format', alt)
 
 
+def convert_imprecise_breakend(std_row: Dict, record: List[VcfRecordType], bp_end: int):
+    """
+    Handles IMPRECISE calls, that leveraged uncertainty from the CIPOS/CIEND/CILEN fields.
+
+    bp1_s = breakpoint1 start
+    bp1_e = breakpoint1 end
+    bp2_s = breakpoint2 start
+    bp2_e = breakpoint2 end
+
+    Insertion and deletion edge case - in which bp1_e > bp2_s
+    E.g bp1_s = 1890, bp1_e = 2000, bp2_s = 1900, bp2_e = 1900.
+    break1 ------------------------=======================--------------
+    break2 ------------------------==========---------------------------
+
+    Insertion edge case - in which bp1_e > bp1_s
+    E.g bp1_s = 1890, bp1_e = 1800, bp2_s = 1800, bp2_e = 1800.
+    break1 ------------------------==-----------------------------------
+    break2 ------------------------=------------------------------------
+
+    Insertion edge case - in which bp1_s > bp2_s
+    E.g bp1_s = 1950, bp1_e = 2000, bp2_s = 1900, bp2_e = 3000.
+    break1 ------------------------==-----------------------------------
+    break2 -----------------------========------------------------------
+    """
+
+    if record.info.get(
+        'CILEN'
+    ):  # as per https://github.com/samtools/hts-specs/issues/615, CILEN takes priority over CIPOS in dictating breakpoint2
+        std_row.update(
+            {
+                COLUMNS.break1_position_start: max(
+                    1, record.pos + record.info.get('CIPOS', (0, 0))[0]
+                ),
+                COLUMNS.break1_position_end: record.pos + record.info.get('CIPOS', (0, 0))[1],
+                COLUMNS.break2_position_start: max(1, bp_end + record.info.get('CILEN', (0, 0))[0]),
+                COLUMNS.break2_position_end: bp_end + record.info.get('CILEN', (0, 0))[1],
+            }
+        )
+    else:
+        std_row.update(
+            {
+                COLUMNS.break1_position_start: max(
+                    1, record.pos + record.info.get('CIPOS', (0, 0))[0]
+                ),
+                COLUMNS.break1_position_end: record.pos + record.info.get('CIPOS', (0, 0))[1],
+                COLUMNS.break2_position_start: max(1, bp_end + record.info.get('CIEND', (0, 0))[0]),
+                COLUMNS.break2_position_end: bp_end + record.info.get('CIEND', (0, 0))[1],
+            }
+        )
+
+    if std_row["break1_chromosome"] == std_row["break2_chromosome"] and (
```

Comment:
style: we do this same check a lot, i might consider setting it to a variable at the start of the function

```python
is_intrachromosomal = std_row["break1_chromosome"] == std_row["break2_chromosome"]


if is_intrachromosomal and ...
```